src/chapters/control-flow/loops.rst

```diff
@@ -0,0 +1,278 @@
+   }
+
+Here, the loop variable moves through the items in the Array of integers, ``k[]``. The syntax
+here uses a colon symbol, ``:``. This type of loop is known as a **for-each loop**.
+
+.. tip::
+
+   When considering this structure, it can be helpful to read the code sample above to yourself
+   as "For each integer in ``Array k``...".
+
+This loop version also works with a String, where we
+can convert the String to an Array of characters:
+
+.. sourcecode:: java
+   :linenos:
+
+   String msg = "Hello World";
+
+   for (char c : msg.toCharArray()) {
+      System.out.println(c);
+   }
+
+As you see, to iterate through a String in this way, Java requires an extra String method,
+``.toCharArray()``, to convert the String to an Array of characters.
+
+.. index:: ! while loop
+
+``while`` Loop
+--------------
+
+Java also supports the **while loop**, or indefinite loop.
+Recall that in Python the ``while`` loop is written as:
+
+A ``while`` loop in Java:
+
+.. sourcecode:: java
+   :linenos:
+
+   while (condition) {
+      statement1
+      statement2
+      ...
+   }
+
+.. index:: ! do-while loop
+
+``do-while`` Loop
+-----------------
+
+Java adds an additional, if seldom used, variation of the ``while`` loop
+called the **do-while loop**. The ``do-while`` loop is very similar to
+``while`` except that the condition is evaluated at the end of the loop
+rather than the beginning. This ensures that a loop *will be executed at
+least one time*. Some programmers prefer this loop in some situations
+because it avoids an additional assignment prior to the loop. 
+
+For example:
+
+.. sourcecode:: java
+   :linenos:
+
+   do {
+      statement1
+      statement2
+      ...
+   } while (condition);
+
+Break Statements in Loops
+-------------------------
+
+There are instances where you may want to terminate a loop if a given
+condition is met. In these instances, the ``break`` statement comes in
+handy. For example, if you want to loop through an array of integers
+searching for a given integer and you want to quit the loop once that
+number is found, you can do the following:
+
+.. sourcecode:: java
+   :linenos:
+
+   public class testBreak {
+
+      public static void main(String [] args) {
+         int[] someInts = {1, 10, 2, 3, 5, 8, 10};
+         int searchTerm = 10;
+         for (int oneInt : someInts) {
+            if (oneInt == searchTerm) {
+               System.out.println("Found it!");
+               break;
+            }
+         }
+      }
+   }
+
+In the code above, instead of the ``for`` loop iterating through all the
+integers in the array, it will stop after it finds the first matching
+instance. So once it finds the first ``10`` in the array, it prints "Found
+it!" and then terminates the loop. If the ``break`` statement weren’t
+there, the loop would continue and when it found the second ``10``, it
+would print "Found it!" a second time.
+
+Note that the ``break`` statement terminates the innermost loop that it
```

Comment:
What do you think about following up with the case where ``break`` occurs in the outer loop?  The result is implied in the current text, but I think we should make the point more concrete.